### ISSUE TITLE
Kernel Panic after restarting UI 

### ISSUE THREAD

**Issue by hanspeteroliver** [GitSquared/edex-ui#943]
Hi! 

First of all, I love eDEX-UI! 

I was adjusting the settings.json and using the "Restart UI" button after editing settings. It happened 2 times that my Mac just turned off in the result of a kernel panic. (freezes for some seconds, puts air vent to 100% for a second and turns off) 

Any idea why this happens? 

Below the log Apple prints In their "Your computer restarted because of a problem" dialog

```
Panic(CPU 1, time 720049727941): NMIPI for spinlock acquisition timeout, spinlock: 0xffffff86cfc868e0, spinlock owner: 0xffffff86d4b5c240, current_thread: 0xffffff86d4b5c240, spinlock_owner_cpu: 0x1
RAX: 0xffffff86d029abf0, RBX: 0xffffff9382fbc968, RCX: 0x0000000000000006, RDX: 0x0000017800000000
RSP: 0xffffffb0dc863aa0, RBP: 0xffffffb0dc863ae0, RSI: 0xffffff86d4b5c240, RDI: 0xffffff9382fbc968
R8:  0xffffffb0dc863c60, R9:  0xffffff8019d10d30, R10: 0x0000000000ffffff, R11: 0x0000000000000000
R12: 0xffffff86d4b5c240, R13: 0x0000000000000000, R14: 0x000001787c549812, R15: 0xffffff86d029abf0
RFL: 0x0000000000000086, RIP: 0xffffff8019cce7aa, CS:  0x0000000000000008, SS:  0x0000000000000010
Backtrace (CPU 1), Frame : Return Address
0xffffffa08906cf80 : 0xffffff8019deee4b 
0xffffffa08906cfd0 : 0xffffff8019c61bdd 
0xffffffb0dc863ae0 : 0xffffff8019cce934 
0xffffffb0dc863b10 : 0xffffff8019d0f550 
0xffffffb0dc863b30 : 0xffffff8019d10d69 
0xffffffb0dc863be0 : 0xffffff8019d11266 
0xffffffb0dc863c50 : 0xffffff8019d102b4 
0xffffffb0dc863d10 : 0xffffff8019d14208 
0xffffffb0dc863d50 : 0xffffff801a2240b4 
0xffffffb0dc863d70 : 0xffffff801a22419e 
0xffffffb0dc863db0 : 0xffffff801a27c93d 
0xffffffb0dc863e30 : 0xffffff801a277e35 
0xffffffb0dc863ee0 : 0xffffff801a277c47 
0xffffffb0dc863f40 : 0xffffff801a369ceb 
0xffffffb0dc863fa0 : 0xffffff8019c621f6 

Process name corresponding to current thread: eDEX-UI Helper (Renderer)
Boot args: chunklist-security-epoch=0 -chunklist-no-rev2-dev chunklist-security-epoch=0 -chunklist-no-rev2-dev

Mac OS version:
20B50

Kernel version:
Darwin Kernel Version 20.1.0: Sat Oct 31 00:07:11 PDT 2020; root:xnu-7195.50.7~2/RELEASE_X86_64
Kernel UUID: 84C6DC45-6B02-335F-9439-5D2A9BC385A4
KernelCache slide: 0x0000000019a00000
KernelCache base:  0xffffff8019c00000
Kernel slide:      0x0000000019a10000
Kernel text base:  0xffffff8019c10000
__HIB  text base: 0xffffff8019b00000
System model name: MacBookPro15,1 (Mac-937A206F2EE63C01)
System shutdown begun: NO
Hibernation exit count: 0

System uptime in nanoseconds: 720049764583
Last Sleep:           absolute           base_tsc          base_nano
  Uptime  : 0x000000a7a64f7ceb
  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000
  Wake    : 0x0000000000000000 0x000000060ee7e7a6 0x0000000000000000
last started kext at 16256440354: >AGPM	119 (addr 0xffffff7fb7e66000, size 118784)
loaded kexts:
>AGPM	119
>X86PlatformShim	1.0.0
>!APlatformEnabler	2.7.0d0
>!ATopCaseHIDEventDriver	4000.27
>!AHIDALSService	1
@filesystems.autofs	3.0
@kext.AMDFramebuffer	4.0.0
@fileutil	20.036.15
@kext.AMDRadeonX4000	4.0.0
@kext.AMDRadeonServiceManager	4.0.0
>!AUpstreamUserClient	3.6.8
>!AGraphicsDevicePolicy	6.1.27
>!AThunderboltIP	4.0.3
@AGDCPluginDisplayMetrics	6.1.27
>pmtelemetry	1
>LuaHardwareAccess	1.0.16
|IOUserEthernet	1.0.1
>usb.!UUserHCI	1
>!A!IKBLGraphics	16.0.0
>BridgeAudioCommunication	100.2
>!AMuxControl2	6.1.27
>!ABridgeAudio!C	100.2
>!AFIVRDriver	4.1.0
>!AAVEBridge	6.1
>!A!ICFLGraphicsFramebuffer	16.0.0
>ACPI_SMC_PlatformPlugin	1.0.0
>!AGFXHDA	100.1.431
|IO!BSerialManager	8.0.1f5
@Dont_Steal_Mac_OS_X	7.0.0
>!AHV	1
>!ADiskImages2	1
>!A!IPCHPMC	2.0.1
@kext.AMD9500!C	4.0.0
>AGDCBacklightControl	6.1.27
>!ABacklight	180.3
>!AMCCSControl	1.14
>!A!ISlowAdaptiveClocking	4.0.0
>BCMWLANFirmware4378.Hashstore	1
>BCMWLANFirmware4377.Hashstore	1
>BCMWLANFirmware4364.Hashstore	1
>BCMWLANFirmware4355.Hashstore	1
>!AFileSystemDriver	3.0.1
@filesystems.tmpfs	1
@filesystems.hfs.kext	556.41.1
@BootCache	40
@!AFSCompression.!AFSCompressionTypeZlib	1.0.0
@!AFSCompression.!AFSCompressionTypeDataless	1.0.0d1
>!ABCMWLANBusInterfacePCIeMac	1
@filesystems.apfs	1677.50.1
@private.KextAudit	1.0
>!ASmartBatteryManager	161.0.0
>!AACPIButtons	6.1
>!ASMBIOS	2.1
>!AACPIEC	6.1
>!AAPIC	1.7
@!ASystemPolicy	2.0.0
@nke.applicationfirewall	310
|IOKitRegistryCompatibility	1
|EndpointSecurity	1
>!AHIDKeyboard	222
>!AHS!BDriver	4000.27
>IO!BHIDDriver	8.0.1f5
>!AActuatorDriver	4400.28
>!AMultitouchDriver	4400.28
>!AInputDeviceSupport	4400.35
@kext.triggers	1.0
@kext.AMDRadeonX4100HWLibs	1.0
@kext.AMDRadeonX4000HWServices	4.0.0
>!AThunderboltEDMSink	5.0.3
>!AThunderboltDPOutAdapter	8.1.4
>!ASMBusPCI	1.0.14d1
|IO!BHost!CUARTTransport	8.0.1f5
|IO!BHost!CTransport	8.0.1f5
>!AGraphicsControl	6.1.27
|IOAccelerator!F2	439.35.4
>IOPlatformPluginLegacy	1.0.0
>X86PlatformPlugin	1.0.0
>!AHDA!C	283.15
|IOHDA!F	283.15
|IOAudio!F	300.6.1
@vecLib.kext	1.2.0
|IOAVB!F	900.12
@plugin.IOgPTPPlugin	900.11
|IOEthernetAVB!C	1.1.0
>IOPlatformPlugin!F	6.0.0d8
@kext.AMDSupport	4.0.0
>!A!ILpssUARTv1	3.0.60
>!A!ILpssUARTCommon	3.0.60
>!AOnboardSerial	1.0
@!AGPUWrangler	6.1.27
@!AGraphicsDeviceControl	6.1.27
>!ABacklightExpert	1.1.0
|IONDRVSupport	585
|IOGraphics!F	585
>!ASMBus!C	1.0.18d1
|IOSlowAdaptiveClocking!F	1.0.0
>usb.cdc.ecm	5.0.0
>usb.cdc.ncm	5.0.0
>usb.IOUSBHostHIDDevice	1.2
>usb.cdc	5.0.0
>usb.networking	5.0.0
>usb.!UHostCompositeDevice	1.2
>!AThunderboltPCIDownAdapter	4.1.1
>!AThunderboltDPInAdapter	8.1.4
>!AThunderboltDPAdapter!F	8.1.4
>!AHPM	3.4.4
>!A!ILpssI2C!C	3.0.60
>!A!ILpssI2C	3.0.60
>!A!ILpssDmac	3.0.60
>!ABSDKextStarter	3
|IOSurface	289.3
@filesystems.hfs.encodings.kext	1
>!ABCMWLANCoreMac	1.0.0
|IOSerial!F	11
|IO80211!FV2	1200.12.2b1
|IOSkywalk!F	1
>mDNSOffloadUserClient	1.0.1b8
>IOImageLoader	1.0.0
>corecapture	1.0.4
>!AXsanScheme	3
>!AThunderboltNHI	7.2.8
|IOThunderbolt!F	9.3.2
>usb.!UVHCIBCE	1.2
>usb.!UVHCICommonBCE	1.0
>usb.!UVHCI	1.2
>usb.!UVHCICommon	1.0
>!AEffaceableNOR	1.0
|IOBufferCopy!C	1.1.0
|IOBufferCopyEngine!F	1
|IONVMe!F	2.1.0
>usb.!UHostPacketFilter	1.0
|IOUSB!F	900.4.2
>usb.!UXHCIPCI	1.2
>usb.!UXHCI	1.2
>!AEFINVRAM	2.1
>!AEFIRuntime	2.1
>!ASMCRTC	1.0
|IOSMBus!F	1.1
|IOHID!F	2.0.0
$!AImage4	3.0.0
|IOTimeSync!F	900.11
|IONetworking!F	3.4
>DiskImages	493.0.0
|IO!B!F	8.0.1f5
|IOReport!F	47
|IO!BPacketLogger	8.0.1f5
$quarantine	4
$sandbox	300.0
@kext.!AMatch	1.0.0d1
|CoreAnalytics!F	1
>!ASSE	1.0
>!AKeyStore	2
>!UTDM	511.40.9
|IOUSBMass!SDriver	184.40.6
|IOSCSIBlockCommandsDevice	436.40.6
|IO!S!F	2.1
|IOSCSIArchitectureModel!F	436.40.6
>!AMobileFileIntegrity	1.0.5
@kext.CoreTrust	1
>!AFDEKeyStore	28.30
>!AEffaceable!S	1.0
>!ACredentialManager	1.0
>KernelRelayHost	1
|IOUSBHost!F	1.2
>!UHostMergeProperties	1.2
>usb.!UCommon	1.0
>!ABusPower!C	1.0
>!ASEPManager	1.0.1
>IOSlaveProcessor	1
>!AACPIPlatform	6.1
>!ASMC	3.1.9
|IOPCI!F	2.9
|IOACPI!F	1.4
>watchdog	1
@kec.pthread	1
@kec.corecrypto	1.0
@kec.Libm	1
panic(cpu 8 caller 0xffffff8019de41c0): "Spinlock acquisition timed out: lock=0xffffff86cfc868e0, lock owner thread=0xffffff86d4b5c240, current_thread: 0xffffff86d029abf0, lock owner active on CPU 0x1, current owner: 0xffffff86d4b5c240, time: 720049802456"@/AppleInternal/BuildRoot/Library/Caches/com.apple.xbs/Sources/xnu/xnu-7195.50.7/osfmk/i386/locks_i386.c:505
Backtrace (CPU 8), Frame : Return Address
0xffffffa089064af0 : 0xffffff8019cbc66d 
0xffffffa089064b40 : 0xffffff8019dff073 
0xffffffa089064b80 : 0xffffff8019def6aa 
0xffffffa089064bd0 : 0xffffff8019c61a2f 
0xffffffa089064bf0 : 0xffffff8019cbbf0d 
0xffffffa089064d10 : 0xffffff8019cbc1f8 
0xffffffa089064d80 : 0xffffff801a4bee1a 
0xffffffa089064df0 : 0xffffff8019de41c0 
0xffffffa089064e30 : 0xffffff8019cdcc36 
0xffffffa089064e60 : 0xffffff8019d0a99b 
0xffffffa089064f20 : 0xffffff8019ddb113 
0xffffffa089064f60 : 0xffffff8019df873a 
0xffffffa089064f80 : 0xffffff8019deee4b 
0xffffffa089064fd0 : 0xffffff8019c61bdd 
0xffffffb0dad03db0 : 0xffffff8019d13ff1 
0xffffffb0dad03e10 : 0xffffff801a279f49 
0xffffffb0dad03f00 : 0xffffff801a279076 
0xffffffb0dad03f40 : 0xffffff801a369ceb 
0xffffffb0dad03fa0 : 0xffffff8019c621f6 

Process name corresponding to current thread: eDEX-UI Helper (Renderer)
```

**Comment by wrac4242** [comment]
hi, which version are you using, are you building from master, or using the latest release?

**Comment by hanspeteroliver** [comment]
I’m using the latest release on macOS 11.0.1

**Comment by wrac4242** [comment]
can you see if there is still an issue running from source?

**Comment by GitSquared** [comment]
> Panic(CPU 1, time 720049727941): NMIPI for spinlock acquisition timeout, spinlock: 0xffffff86cfc868e0

This is very low-level (meaning: it's not something my code handles) and i'm not sure what could be causing this.

If you don't use the "reload UI" button, everything works fine?
Could you give me the exact model of the Mac?

**Comment by GitSquared** [comment]
Closed as stale - see v2.2.6 for experimental apple silicon support which might help, too
